### PR TITLE
feat: support following shows and fairs

### PR DESCRIPTION
**Pull request by brainbicycle**
This PR resolves [] 

### Description

Adds follow buttons to show and fair page. 

follow-show

follow-fair

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I hid my changes behind a **[feature flag]**, or they don't need one.
- [ ] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- support following shows and fairs - brian 

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (support following shows and fairs - brian)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d43150de395f564537adf486426082ed3ac53ff0
</p>

File: src/app/Scenes/Fair/Components/FairHeader.tsx [hunk 1]
```diff
@@ -11,16 +11,33 @@ export const FairHeader: FC<FairHeaderProps> = ({ fair }) => {
...
+  const handleFollowFair = () => {
+    if (!data.profile) {
+      return
+    }
+
+    commit({
+      variables: {
+        input: {
+          profileID: data.profile.internalID,
+          unfollow: data.profile.isFollowed,
```
Comment (araujobarret) [inline src/app/Scenes/Fair/Components/FairHeader.tsx:31]: I don't think we change the follow in real-time so you need some optimistic response here to simulate the change in real-time, [example](https://github.com/artsy/eigen/blob/190e3af140b171d1b55519fdc28a2c04bb79cf38/src/app/Components/Artist/useFollowArtist.ts#L40-L41)

File: src/app/Scenes/Fair/Components/FairHeader.tsx [hunk 2]
```diff
@@ -51,14 +68,39 @@ export const FairHeader: FC<FairHeaderProps> = ({ fair }) => {
...
+          onPress={handleFollowFair}
+          loading={isInFlight}
+          disabled={isInFlight}
+        />
+      </Flex>
     </Flex>
   )
 }
 
+const FollowFairMutation = graphql`
```
Comment (araujobarret) [inline src/app/Scenes/Fair/Components/FairHeader.tsx:86]: in general, we'd put the mutation in `Fair/mutations/useFollowFair.ts` and only import it from there

File: src/app/Scenes/Show/Components/ShowHeader.tsx [hunk 3]
```diff
@@ -12,12 +12,32 @@ export interface ShowHeaderProps extends BoxProps {
...
 
+  const handleFollowShow = () => {
+    commit({
+      variables: {
+        input: {
+          partnerShowID: show?.internalID,
+          unfollow: !!show?.isFollowed,
+        },
+      },
+      // TODO: handle errors? we don't seem to handle elsewhere :(
```
Comment (araujobarret) [inline src/app/Scenes/Show/Components/ShowHeader.tsx:31]: ditto about the optimistic response

File: src/app/Scenes/Show/Components/ShowHeader.tsx [hunk 4]
```diff
@@ -44,6 +67,14 @@ export const ShowHeader: React.FC<ShowHeaderProps> = ({ show, ...rest }) => {
         </Text>
       )}
 
+      <FollowButton
+        haptic
+        isFollowed={!!show.isFollowed ? show.isFollowed : false}
```
Comment (araujobarret) [inline src/app/Scenes/Show/Components/ShowHeader.tsx:72]: I think here it's just `isFollowed={!!show.isFollowed}`

File: src/app/Scenes/Show/Components/ShowHeader.tsx [hunk 5]
```diff
@@ -12,12 +12,32 @@ export interface ShowHeaderProps extends BoxProps {
...
+  const handleFollowShow = () => {
+    commit({
+      variables: {
+        input: {
+          partnerShowID: show?.internalID,
+          unfollow: !!show?.isFollowed,
+        },
+      },
+      // TODO: handle errors? we don't seem to handle elsewhere :(
+      // onError(error) {
```
Comment (araujobarret) [inline src/app/Scenes/Show/Components/ShowHeader.tsx:32]: we could add a simple toast as we do in other areas  with `useToast`, better than nothing or waiting for any design input :

**Comment by brainbicycle** [comment]
closing for now but can be used as a base if we want to pursue!

**Pull request closed without merging**